bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -80,11 +80,15 @@ type KubeadmConfigSpec struct {
 	// preKubeadmCommands specifies extra commands to run before kubeadm runs
 	// +optional
 	// +kubebuilder:validation:MaxItems=1000
+	// +kubebuilder:validation:items:MinLength=1
+	// +kubebuilder:validation:items:MaxLength=4096
```

Comment:
Not sure this is enough, e.g. https://github.com/kubernetes-sigs/cluster-api-provider-packet/blob/988c62dc28fac0b66ebd5032769f5d1f40c0ca20/templates/cluster-template-crs-cni.yaml#L33 has >2k chars already. 

But ok to start with this. Such big scripts should go to a file entry and be executed as a command here only.

No strong opinion..

Reply:
Yeah. That's my thinking. Large scripts should go in a file not in a command. I bump to reduce the chances of breaking folks

Reply:
+1 to go big here, folks are using kubeadmCommands as a swiss army knife